### PR TITLE
Feature 1635 - added labels to generate HPA

### LINKED ISSUE #1631: [Feature]: Support HPA configuration
### What would you like to be added?

I would like to introduce scalability features, by working on the basic version of the HPA, `minReplicas`, `maxReplicas`, `cpu`, and `memory` metrics.

### Why is this needed?

Make users' life easier when it comes to deploying production-ready applications, using the scalability features.

### PR DESCRIPTION
**Pull request by sosan**
#### What type of PR is this?

/kind feature 

#### What this PR does / why we need it:
added this labels to min/max replicas and custom metrics for cpu and memory:

kompose.hpa.minreplicas
kompose.hpa.maxreplicas
kompose.hpa.cpu
kompose.hpa.memory

documentated labels user_guide.md

generate custom metrics from cpu, memory and set min/max replicas

#### Which issue(s) this PR fixes:

Fixes #1631

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -467,6 +471,63 @@ services:
     labels:
       kompose.volume.sub-path: pg-data
 ```
+
+- `kompose.hpa.minreplicas` defines minimum replicas from Horizontal Pod Autoscaler. Default value 1 [HPA Min Replicas](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics).
+
+For example:
+
+```yaml
+version: '3.8'
```
Comment (cdrage) [inline docs/user-guide.md:480]: needs to remove version

File: docs/user-guide.md [hunk 2]
```diff
@@ -467,6 +471,63 @@ services:
...
+    labels:
+      kompose.hpa.minreplicas: 1
+```
+
+- `kompose.hpa.maxreplicas` defines maximum replicas from Horizontal Pod Autoscaler. Default value 10 [HPA Max Replicas](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics).
+
+For example:
+
+```yaml
+version: '3.8'
```
Comment (cdrage) [inline docs/user-guide.md:494]: needs to remove version

File: docs/user-guide.md [hunk 3]
```diff
@@ -467,6 +471,63 @@ services:
...
+    labels:
+      kompose.hpa.maxreplicas: 10
+```
+
+- `kompose.hpa.cpu` defines % cpu utilization trigger scale from Horizontal Pod Autoscaler. It is represented as a percentage of a resource. Default value: 50 [HPA CPU Utilization](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics).
+
+For example:
+
+```yaml
+version: '3.8'
```
Comment (cdrage) [inline docs/user-guide.md:508]: needs to remove version

File: docs/user-guide.md [hunk 4]
```diff
@@ -467,6 +471,63 @@ services:
...
+    labels:
+      kompose.hpa.cpu: 50
+```
+
+- `kompose.hpa.memory` defines memory utilization trigger scale from Horizontal Pod Autoscaler. It is represented as a percentage of a resource. Default value: 70 [HPA Memory Utilization](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics).
+
+For example:
+
+```yaml
+version: '3.8'
```
Comment (cdrage) [inline docs/user-guide.md:522]: needs to remove version

File: docs/user-guide.md [hunk 5]
```diff
@@ -211,6 +211,10 @@ The currently supported options are:
 | kompose.cronjob.schedule                            | kubernetes cronjob schedule (for example: '1 * * * *')                               |
 | kompose.cronjob.concurrency_policy                  | 'Forbid' / 'Allow' / 'Never' / ''                                                    |
 | kompose.cronjob.backoff_limit                       | kubernetes cronjob backoff limit (for example: '6')                                  |
+| kompose.hpa.minreplicas                             | defines Horizontal Pod Autoscaler min pod replicas                                   |
+| kompose.hpa.maxreplicas                             | defines Horizontal Pod Autoscaler max pod replicas                                   |
```
Comment (cdrage) [inline docs/user-guide.md:215]: we need to follow how we do the other labels.

this should be instead: `kompose.hpa.replicas.min` and `kompose.hpa.replicas.max`

**Review by cdrage — CHANGES_REQUESTED** [review]
This is amazing work, great job on the implementation!

Needs e2e and regular tests though, sorry!

**Review by TessaIO — COMMENTED** [review]
Thanks @sosan for this amazing enhancement. I left a review, and I'll do another round when you finish this!

**Comment by sosan** [comment]
done :ok_hand:

**Comment by sosan** [comment]
fixed merge conflicts

**Review by cdrage — APPROVED** [review]
Amazing, thanks for the awesome work! All the code LGTM!

**Pull request merged**